apps/mobile/src/components/TradeCalculator/atoms.ts

```diff
@@ -65,47 +52,49 @@ export const btcPriceForOfferWithStateAtom =
 export const applyFeeOnFeeChangeActionAtom = atom(
   null,
   (get, set, newFee: number) => {
-    const btcOrSat = get(btcOrSatAtom)
-    const btcInputValue = Number(get(btcInputValueAtom))
+    const fiatInputValue = Number(get(fiatInputValueAtom))
     const previousAppliedFee = get(feeAmountAtom)
 
-    if (get(btcInputValueAtom)) {
-      const btcValueWithoutPreviousFee = cancelFeeOnBtcAmount(
-        btcInputValue,
+    if (get(fiatInputValueAtom)) {
+      const fiatValueWithoutPreviousFee = cancelFeeOnNumberValue(
+        fiatInputValue,
         previousAppliedFee
       )
-      const btcValueWithNewFeeApplied = applyFeeOnBtcAmount(
-        btcValueWithoutPreviousFee,
+      const fiatValueWithNewFeeApplied = applyFeeOnNumberValue(
+        fiatValueWithoutPreviousFee,
         newFee
       )
 
-      set(
-        btcInputValueAtom,
-        btcOrSat === 'SAT'
-          ? String(Math.round(btcValueWithNewFeeApplied))
-          : removeTrailingZerosFromNumberString(
-              btcValueWithNewFeeApplied.toFixed(DECIMALS_FOR_BTC_VALUE)
-            )
-      )
+      set(fiatInputValueAtom, String(Math.round(fiatValueWithNewFeeApplied)))
     }
     set(feeAmountAtom, newFee)
   }
 )
 
+export const saveYourPriceActionAtom = atom(null, (get, set) => {
+  const ownPrice = get(ownPriceAtom)
+
+  if (ownPrice) {
+    set(tradeBtcPriceAtom, Number(ownPrice))
+  }
+  set(calculateFiatValueOnBtcAmountChangeActionAtom, {
+    btcAmount: get(btcInputValueAtom),
+  })
+
+  set(tradePriceTypeAtom, 'your')
+  // set(applyFeeOnTradePriceTypeChangeActionAtom)
+})
+
 export const applyFeeOnTradePriceTypeChangeActionAtom = atom(
   null,
   (get, set) => {
-    const btcInputValue = Number(get(btcInputValueAtom))
+    const fiatInputValue = Number(get(fiatInputValueAtom))
```

Comment:
Can it happen that the number cast fails? If so what will happen in the app?

Reply:
I checked it, this should be OK, as input value is checked for non numeric chars on each change and in the case of occurance is the value of the input replaced with zero string, like '0'